numba_dpex/experimental/_kernel_dpcpp_spirv_overloads/_atomic_ref_overloads.py

```diff
@@ -209,6 +209,157 @@ def codegen(context, builder, sig, args):
+
+        spirv_fn_arg_types = [
+            ptr_type,
+            llvmir.IntType(32),
+            llvmir.IntType(32),
+        ]
+
+        mangled_fn_name = ext_itanium_mangler.mangle_ext(
+            "__spirv_AtomicLoad",
+            [
+                types.CPointer(atomic_ref_dtype, addrspace=ptr_type.addrspace),
+                "__spv.Scope.Flag",
+                "__spv.MemorySemanticsMask.Flag",
+            ],
+        )
+
+        fn = cgutils.get_or_insert_function(
+            builder.module,
+            llvmir.FunctionType(retty, spirv_fn_arg_types),
+            mangled_fn_name,
+        )
+        fn.calling_convention = CC_SPIR_FUNC
+        spirv_memory_semantics_mask = get_memory_semantics_mask(
+            atomic_ref_ty.memory_order
+        )
+        spirv_scope = get_scope(atomic_ref_ty.memory_scope)
+
+        fn_args = [
+            builder.extract_value(args[0], data_attr_pos),
+            context.get_constant(types.int32, spirv_scope),
+            context.get_constant(types.int32, spirv_memory_semantics_mask),
+        ]
+
+        return builder.call(fn, fn_args)
+
+    return sig, _intrinsic_load_gen
+
+
+def _store_exchange_intrisic_helper(context, builder, sig, ol_info: dict):
+    atomic_ref_ty = sig.args[0]
+    atomic_ref_dtype = atomic_ref_ty.dtype
+
+    ptr_type = context.get_value_type(atomic_ref_dtype).as_pointer()
+    ptr_type.addrspace = atomic_ref_ty.address_space
+
+    spirv_fn_arg_types = [
+        ptr_type,
+        llvmir.IntType(32),
+        llvmir.IntType(32),
+        context.get_value_type(atomic_ref_dtype),
+    ]
+
+    mangled_fn_name = ext_itanium_mangler.mangle_ext(
+        ol_info["name"],
+        [
+            types.CPointer(atomic_ref_dtype, addrspace=ptr_type.addrspace),
+            "__spv.Scope.Flag",
+            "__spv.MemorySemanticsMask.Flag",
+            atomic_ref_dtype,
+        ],
+    )
+
+    fn = cgutils.get_or_insert_function(
+        builder.module,
+        llvmir.FunctionType(ol_info["retty"], spirv_fn_arg_types),
+        mangled_fn_name,
+    )
+    fn.calling_convention = CC_SPIR_FUNC
+
+    fn_args = [
+        builder.extract_value(
+            ol_info["args"][0],
+            context.data_model_manager.lookup(atomic_ref_ty).get_field_position(
+                "ref"
+            ),
+        ),
+        context.get_constant(
+            types.int32, get_scope(atomic_ref_ty.memory_scope)
+        ),
+        context.get_constant(
+            types.int32, get_memory_semantics_mask(atomic_ref_ty.memory_order)
+        ),
+        ol_info["args"][1],
+    ]
+
+    return builder.call(fn, fn_args)
+
+
+@intrinsic(target=DPEX_KERNEL_EXP_TARGET_NAME)
+def _intrinsic_store(
+    ty_context, ty_atomic_ref, ty_val
+):  # pylint: disable=unused-argument
+    sig = types.void(ty_atomic_ref, ty_val)
+
+    def _intrinsic_store_gen(context, builder, sig, args):
+        _store_exchange_intrisic_helper(
```

Comment:
Why name `store_exchange_intrinsic_helper` v/s `store_intrinsic_helper`?

Reply:
It is used by both store and exchange functions